src/Forms/Shared/Samples/Network Analysis/ConnectedTrace/ConnectedTrace.xaml.cs

```diff
@@ -0,0 +1,320 @@
+                UtilityElement element = null;
+
+                // Select default terminal or display possible terminals for the junction feature.
+                UtilityNetworkSource networkSource = _utilityNetwork.Definition.GetNetworkSource(feature.FeatureTable.TableName);
+
+                // Check if the network source is a junction or an edge.
+                if (networkSource.SourceType == UtilityNetworkSourceType.Junction)
+                {
+                    // Get the UtilityAssetGroup from the feature.
+                    string assetGroupFieldName = ((ArcGISFeatureTable)feature.FeatureTable).SubtypeField ?? "ASSETGROUP";
+                    int assetGroupCode = Convert.ToInt32(feature.Attributes[assetGroupFieldName]);
+                    UtilityAssetGroup assetGroup = networkSource?.AssetGroups?.FirstOrDefault(g => g.Code == assetGroupCode);
+
+                    // Get the UtilityAssetType from the feature.
+                    int assetTypeCode = Convert.ToInt32(feature.Attributes["ASSETTYPE"]);
+                    UtilityAssetType assetType = assetGroup?.AssetTypes?.FirstOrDefault(t => t.Code == assetTypeCode);
+
+                    // Get the list of terminals for the feature.
+                    IEnumerable<UtilityTerminal> terminals = assetType?.TerminalConfiguration?.Terminals;
+
+                    // If there is more than one terminal, prompt the user to select a terminal.
+                    if (terminals.Count() > 1)
+                    {
+                        // Ask the user to choose the terminal.
+                        UtilityTerminal terminal = await WaitForTerminal(terminals);
+
+                        // Create a UtilityElement with the terminal.
+                        element = _utilityNetwork.CreateElement(feature, terminal);
+                        Status.Text = $"terminal: `{terminal?.Name ?? "default"}`";
+                    }
+                    else
+                    {
+                        element = _utilityNetwork.CreateElement(feature, terminals.FirstOrDefault());
+                        Status.Text = $"terminal: `{element.Terminal?.Name ?? "default"}`";
+                    }
+                }
+                else if (networkSource.SourceType == UtilityNetworkSourceType.Edge)
+                {
+                    element = _utilityNetwork.CreateElement(feature);
+
+                    // Compute how far tapped location is along the edge feature.
+                    if (feature.Geometry is Polyline line)
+                    {
+                        line = GeometryEngine.RemoveZ(line) as Polyline;
+
+                        // Set how far the element is along the edge.
+                        element.FractionAlongEdge = GeometryEngine.FractionAlong(line, e.Location, -1);
+
+                        Status.Text = $"fractionAlongEdge: `{element.FractionAlongEdge}`";
```

Comment:
How is the fraction along relevant to the trace operation? Does it need to be shown to the user or is this an implementation detail?

Reply:
FractionAlongEdge is an attribute of the UtilityElement object. We show the user the value set for that attribute. I can reword the status to be more user friendly and appear less like a debug output.